30_regexp_lab/relab.txt

```diff
@@ -1,17 +1,19 @@
 Use grep to print only the lines from "input" for which:
 
 1. There is a binary number in the line
+	cat input |grep -a [0-9]
 
 2. There is an even binary number in the line
-
+	cat input |grep -a *[0-9]
```

Comment:
An even binary number is a binary number that ends in 0. Given the answer of (1), the correct solution here is `grep -w [01]*0`. Its meaning:

Search for a sequence of 0 or more characters from group (0,1), followed by a single 0, and as its own word.